### PR TITLE
Make sure that GitRepositoryProvider.repositoryForDirectorySync() return...

### PR DESCRIPTION
**Pull request by bolinfest**
...s null rather than throws.

The UI locks up if this method does not return.

### CODE REVIEW

**Comment by bolinfest** [comment]
Unit test forthcoming.

**Comment by bolinfest** [comment]
Unit test provided. There's a `TODO` about logging the error. I'm not sure what's standard practice in Atom for something like this.

File: spec/git-repository-provider-spec.coffee [hunk 1]
```diff
@@ -54,3 +54,35 @@ describe "GitRepositoryProvider", ->
           directory = new Directory dirPath
           provider.repositoryForDirectory(directory).then (result) ->
             expect(result).toBe null
+
+    describe "when specified a Directory without existsSync()", ->
+      it "returns null", ->
+        provider = new GitRepositoryProvider atom.project
+
+        # Tolerate an implementation of Directory that does not implement existsSync().
```
Comment (maxbrunsfeld) [inline spec/git-repository-provider-spec.coffee]: Could you put the code that sets up `directory` into a `beforeEach`. My criteria for extracting something like that is roughly:
* It's the same for all of the tests in the `describe` block.
* It corresponds directly to the description of the `describe` block.

File: src/git-repository-provider.coffee [hunk 2]
```diff
@@ -26,9 +26,9 @@ isValidGitDirectorySync = (directory) ->
   # To decide whether a directory has a valid .git folder, we use
   # the heuristic adopted by the valid_repository_path() function defined in
   # node_modules/git-utils/deps/libgit2/src/repository.c.
-  return directory.getSubdirectory('objects').existsSync() and
-      directory.getFile('HEAD').existsSync() and
-      directory.getSubdirectory('refs').existsSync()
+  return directory.getSubdirectory('objects').existsSync?() and
```
Comment (maxbrunsfeld) [inline src/git-repository-provider.coffee]: I would say leave `isValidGitDirectorySync` as it was; we only need to check once if the `Directory` (and the directories it returns) implements synchronous IO; no need to check each call.

File: spec/git-repository-provider-spec.coffee (hunk 1 above)
Comment (bolinfest) [inline spec/git-repository-provider-spec.coffee]: Done.

File: src/git-repository-provider.coffee (hunk 2 above)
Comment (bolinfest) [inline src/git-repository-provider.coffee]: Done.

**Pull request merged**